### PR TITLE
Add Feature/seed database and General Cleanup

### PR DESCRIPTION
**Pull request by ErikPlachta**
# Add Seed Data #20, verified the integrity of Models with Seed creation, changed Table `Character` to `Hero`, and clean-up overall in the source content I provided to correct.

---

### Seed Data Onboarded #20 

- [x] Development team can onboard seed data with bash command `npm run seed`
- [x] Fully using Models created by @mmlawton15 on issue #15 

**Seeding works with `npm run seed`**
image

**Data in database local database**
image

---

### Fixed `RENAME.env` variable names

- [x] Updated env variables in `RENAME.env` to match what everything is looking for.

### Table Name Change from `Character` to `Hero` to address MySQL Conflict with name `Character`

[x] Had to rename the Model `Character` to something else because MySQL did NOT like that as a table name.
   - [x] Changed from `Character` to `Hero`.  
   - Related to #15 

---

### README

- [x] Updated README with some base content to start working on #14 

---

## SEED DATA FOR AWARENESS

**User**
```json
[
    {
        "id": 1,
        "username": "Christina",
        "email" : "christina@noemail.com",
        "password": "christina_123"
    },
    {
        "id": 2,
        "username": "Connie",
        "email" : "connie@noemail.com",
        "password": "connie_123"
    },
    {
        "id": 3,
        "username": "Erik",
        "email" : "erik@noemail.com",
        "password": "erik_123"
    },
    {
        "id": 4,
        "username": "Mary",
        "email" : "mary@noemail.com",
        "password": "mary_123"
    }
]
```

**Hero**

```json
[
    {
        "id": 1,
        "user_id": 1,
        "name" : "Christina",
        "race": "Human",
        "class": "Ranger",
        "gender": "Female"
    },
    {
        "id": 2,
        "user_id": 2,
        "name" : "Connie",
        "race": "Human",
        "class": "Elf",
        "gender": "Female"
    },
    {
        "id": 3,
        "user_id": 3,
        "name" : "Erik",
        "race": "Drawrf",
        "class": "Warrior",
        "gender": "Male"
    },
    {
        "id": 4,
        "user_id": 4,
        "name" : "Mary",
        "race": "Human",
        "class": "Paladin",
        "gender": "Female"
    }
]
```

### CODE REVIEW

File: config/connection.js [hunk 1]
```diff
@@ -15,9 +15,9 @@ if (process.env.JAWSDB_URL) {
   sequelize = new Sequelize(
     process.env.DB_NAME,
     process.env.DB_USER,
-    process.env.DB_PW,
+    process.env.DB_PASSWORD,
```
Comment (ErikPlachta) [inline config/connection.js:18]: Fixed env variable names here to mirror what ORM / Controller content is looking for.

File: config/connection.js [hunk 2]
```diff
@@ -15,9 +15,9 @@ if (process.env.JAWSDB_URL) {
...
     {
-      host: 'localhost',
+      host: process.env.SERVER_PATH,
```
Comment (ErikPlachta) [inline config/connection.js:20]: Updated host to use an env variable

File: controllers/api/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+// // -- Giving handlebar routing params to Router
+// router.use('/', homeRoutes);
+
+//-- if gets here when rounting, throw 404
+router.use((req, res) => {
+  res.status(404).end();
+});
+
+
+module.exports = router;
```
Comment (ErikPlachta) [inline controllers/api/index.js:17]: Built this so that Controller, _Express.js_, could run properly.

File: controllers/index.js [hunk 4]
```diff
@@ -1,16 +1,10 @@
...
 
-// // -- Giving handlebar routing params to Router
-// router.use('/', homeRoutes);
-
-//-- if gets here when rounting, throw 404
-router.use((req, res) => {
-  res.status(404).end();
-});
-
-module.exports = router;
```
Comment (ErikPlachta) [inline controllers/index.js:16]: Cleaned up the unused template code I provided.

File: controllers/index.js [hunk 5]
```diff
@@ -1,16 +1,10 @@
+//-- Import Express and create router to share existing express instance
 const router = require('express').Router();
-//const apiRoutes = require('./api');
-// -- Adding handlebars routing
-// const homeRoutes = require('./home-routes.js');
```
Comment (ErikPlachta) [inline controllers/index.js:4]: Cleaned up the unused template code I provided.

File: package.json [hunk 6]
```diff
@@ -1,25 +1,38 @@
...
-  "author": "",
-  "license": "ISC",
-  "bugs": {
-    "url": "https://github.com/barrantesc/The-Sheet-Show/issues"
-  },
-  "homepage": "https://github.com/barrantesc/The-Sheet-Show#readme",
-  "dependencies": {
-    "express": "^4.17.2",
-    "mysql2": "^2.3.3",
-    "sequelize": "^6.16.0"
```
Comment (ErikPlachta) [inline package.json:23]: Imported `package.json`  and verified integrity

File: seeds/connection_sequelize.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  {
+    host: process.env.SERVER_PATH,
+    dialect: 'mysql',
+    port: 3306
+  },
+  console.log(`//-- Connection MYSQL database with npm package Sequelize success!`)
+);
+
+//-- exporting created sequelize obj
+module.exports = sequelize;
```
Comment (ErikPlachta) [inline seeds/connection_sequelize.js:19]: Added this file here as a temp solution knowing it is not the best way to handle, but allowed it to work
> We'll need to point it to connection later!

File: seeds/seed.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    .catch(console.log)
+};
+
+//------------------------------------------------------------------------------
+//-- RUNNING 
+
+seed();
+
+// seedDatabase();
+
```
Comment (ErikPlachta) [inline seeds/seed.js:94]: Created a script to onboard seed data.
> It uses MySQL2 and Sequelize together.

**Review by ErikPlachta — COMMENTED** [review]
I reviewed, added comments, and verified all change integrity. 
> Waiting for team to review and someone to merge.

**Review by NicaVulcan — APPROVED** [review]
:)

**Comment by ErikPlachta** [comment]
## Full transparency - These usernames will be purged before cloud deploy and not used.

**Comment by ErikPlachta** [comment]
## Some linked issues that are not closed are actually completed but just need to be reviewed, updated, and then closed.

**Pull request merged**